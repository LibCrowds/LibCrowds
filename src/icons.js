import Vue from 'vue'
import Icon from 'vue-awesome/components/Icon'
import users from 'ionicons/dist/svg/ios-people-outline.svg'
import projects from 'ionicons/dist/svg/ios-desktop-outline.svg'
import taskruns from 'ionicons/dist/svg/ios-create-outline.svg'

Vue.component('icon', Icon)

Icon.register({
  'users': {
    width: 512,
    height: 512,
    raw: users
  }
})

Icon.register({
  'taskruns': {
    width: 512,
    height: 512,
    raw: taskruns
  }
})

Icon.register({
  'projects': {
    width: 512,
    height: 512,
    raw: projects
  }
})
