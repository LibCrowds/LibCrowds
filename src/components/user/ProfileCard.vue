<template>
  <b-card class="user-card" no-block>

    <div class="avatar-background">
      <user-avatar
        :user="user"
        :size="200">
      </user-avatar>
    </div>

    <div id="user-details" class="card-block">
      <div class="card-title">
        <h2>{{ user.name }}</h2>
      </div>
      <div class="card-text">
        <ul>
          <li>
            <icon name="trophy"></icon>Rank {{ user.rank }}
          </li>
          <li>
            <icon name="tasks"></icon>Contributions: {{ user.n_answers }}
          </li>
        </ul>
      </div>
    </div>
    <div class="card-block">
      <slot></slot>
    </div>
  </div>

  </b-card>
</template>

<script>
import 'vue-awesome/icons/trophy'
import 'vue-awesome/icons/tasks'
import 'vue-awesome/icons/clock-o'
import UserAvatar from '@/components/user/Avatar'

export default {
  props: {
    user: {
      type: Object,
      required: true
    }
  },

  components: {
    UserAvatar
  }
}
</script>

<style lang="scss" scoped>
@import 'src/assets/style/main';

.card.user-card {
  .avatar-background {
    padding-top: $spacer-y * 2;
    padding-left: $spacer-x * 2;
    padding-right: $spacer-x * 2;
    background:
      linear-gradient(to bottom, rgba($brand-info, 0) 0%,
      rgba($brand-info, 0) calc(50% + (#{$spacer-y} * 1)),
      $white calc(50% + (#{$spacer-y} * 1)), $white 100% ),
      url('../../assets/img/background.jpg');
    background-size: cover;
    background-position: center center;
  }

  #user-details {
    text-align: center;

    ul {
      list-style: none;
      padding: 0;
    }

    li {
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: center;
      margin-bottom: 0.5rem;

      svg {
        margin-right: 5px;
      }
    }
  }
}
</style>
