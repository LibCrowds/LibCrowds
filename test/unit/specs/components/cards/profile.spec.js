import BootstrapVue from 'bootstrap-vue'
import Icon from 'vue-awesome/components/Icon'
import 'vue-awesome/icons/trophy'
import 'vue-awesome/icons/tasks'
import 'vue-awesome/icons/clock-o'

import pbTestResponses from '~/test/fixtures/pybossaTestResponses.json'
import { mount, createLocalVue } from 'vue-test-utils'
import ProfileCard from '~/components/cards/Profile'

describe('Profile card', () => {
  let localVue = null
  let user = null
  let wrapper = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    localVue.component('icon', Icon)
    user = pbTestResponses.getAccount.data.user
    wrapper = mount(ProfileCard, {
      localVue,
      propsData: {
        user: user
      }
    })
  })

  it('renders correctly', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
