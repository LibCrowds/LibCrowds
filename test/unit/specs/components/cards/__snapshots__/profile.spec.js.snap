// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile card renders correctly 1`] = `

<div data-server-rendered="true"
     class="profile-card card"
>
  <div class="avatar-background">
    <div class="user-avatar mx-auto"
         style="width:200px;"
    >
      <v-gravatar triggers
                  title="JohnDoe"
                  email="JohnDoe"
                  size="200"
                  default-img="identicon"
                  alt="Avatar for JohnDoe"
                  class="img-thumbnail rounded-circle"
      >
      </v-gravatar>
    </div>
  </div>
  <div id="user-details"
       class="card-body"
  >
    <div class="card-title">
      <h2>
        JohnDoe
      </h2>
    </div>
    <div class="card-text">
      <ul>
        <li>
          <svg version="1.1"
               role="presentation"
               width="14.857142857142858"
               height="16"
               viewbox="0 0 1664 1792"
               class="fa-icon"
          >
            <path d="M458 883q-74-162-74-371h-256v96q0 78 94.5 162t235.5 113zM1536 608v-96h-256q0 209-74 371 141-29 235.5-113t94.5-162zM1664 480v128q0 71-41.5 143t-112 130-173 97.5-215.5 44.5q-42 54-95 95-38 34-52.5 72.5t-14.5 89.5q0 54 30.5 91t97.5 37q75 0 133.5 45.5t58.5 114.5v64q0 14-9 23t-23 9h-832q-14 0-23-9t-9-23v-64q0-69 58.5-114.5t133.5-45.5q67 0 97.5-37t30.5-91q0-51-14.5-89.5t-52.5-72.5q-53-41-95-95-113-5-215.5-44.5t-173-97.5-112-130-41.5-143v-128q0-40 28-68t68-28h288v-96q0-66 47-113t113-47h576q66 0 113 47t47 113v96h288q40 0 68 28t28 68z">
            </path>
            
            <!---->
          </svg>
          Rank 1813
        </li>
        <li>
          <svg version="1.1"
               role="presentation"
               width="16"
               height="16"
               viewbox="0 0 1792 1792"
               class="fa-icon"
          >
            <path d="M1024 1408h640v-128h-640v128zM640 896h1024v-128h-1024v128zM1280 384h384v-128h-384v128zM1792 1216v256q0 26-19 45t-45 19h-1664q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h1664q26 0 45 19t19 45zM1792 704v256q0 26-19 45t-45 19h-1664q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h1664q26 0 45 19t19 45zM1792 192v256q0 26-19 45t-45 19h-1664q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h1664q26 0 45 19t19 45z">
            </path>
            
            <!---->
          </svg>
          Contributions: 56
        </li>
      </ul>
    </div>
  </div>
  <div class="card-body">
  </div>
</div>

`;
